Create s12cpuv2-mode

AdamNiederer opened the pull request:
### Brief summary of what the package does

A major mode for the assembly language of the S12CPUV2 found on some Freescale microcontrollers

### Direct link to the package repository

https://github.com/AdamNiederer/s12cpuv2-mode

### Your association with the package

Wrote it

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
Thanks. Quick feedback:
- Don't override `TAB` and `<backtab>`: instead, set your own `indent-line-function`, then all the standard Emacs machinery will work as the user expects, however they have customised it.
- `C-c l` is reserved for end users, and must not be used here. Recent versions of `package-lint` warn about this.
- You're using `setq-local`, which only appeared in Emacs 24.3, so you'll need to depend on that Emacs version at a minimum. `package-lint` should also have told you about this.
Looks good otherwise!

AdamNiederer commented:
Thanks for the feedback; I updated package-lint and fixed the issues.
Quick question: It doesn't look like `<backtab>` is bound to anything in `prog-mode`. Does binding it to something do anything unexpected? Removing the binding breaks one's ability to unindent lines with `<backtab>` from what I can see.

purcell commented:
> Quick question: It doesn't look like <backtab> is bound to anything in prog-mode. Does binding it to something do anything unexpected? Removing the binding breaks one's ability to unindent lines with <backtab> from what I can see.

You're right: I think it's reasonable in this case. I found a bunch of other packages which define `backtab` locally.

tarsius commented:
> You're right: I think it's reasonable in this case.

@purcell So we should merge?

purcell commented:
Yep. Thanks for your patience, @AdamNiederer.
